### PR TITLE
vmm: Add fallback handling for sending live migration

### PR DESCRIPTION
**Pull request by likebreath**
This patch adds a fallback path for sending live migration, where it
ensures the following behavior of source VM post live-migration:

1. The source VM will be paused only when the migration is completed
successfully, or otherwise it will keep running;

2. The source VM will always stop dirty pages logging.

Fixes: #2895

Signed-off-by: Bo Chen <chen.bo@intel.com>

### CODE REVIEW

File: vmm/src/lib.rs [hunk 1]
```diff
@@ -999,141 +999,161 @@ impl Vmm {
...
-                Request::abandon().write_to(&mut socket)?;
-                Response::read_from(&mut socket).ok();
-                return Err(MigratableError::MigrateSend(anyhow!(
-                    "Error completing migration"
-                )));
+                    Ok(())
+                }
+                // Ensure the source VM continue to run upon unsuccessful migration
+                Err(e) => {
+                    // Stop logging dirty pages
```
Comment (rbradford) [inline vmm/src/lib.rs:1147]: Probably should put `error!("Migration failed: {}", e)` in here to match the above "Migration complete"

File: vmm/src/lib.rs [hunk 2]
```diff
@@ -999,141 +999,161 @@ impl Vmm {
...
+                    vm.stop_memory_dirty_log()?;
+
+                    if paused {
```
Comment (rbradford) [inline vmm/src/lib.rs:1150]: Is there a function on `vm` that you can use to look at the state to avoid this bool.

**Review by rbradford — APPROVED** [review]

File: vmm/src/lib.rs (hunk 1 above)
Comment (likebreath) [reply vmm/src/lib.rs:1147]: Good point. Added.

File: vmm/src/lib.rs (hunk 2 above)
Comment (likebreath) [reply vmm/src/lib.rs]: Right. I think `vm.get_state()` should do the work. PR updated.

**Pull request merged**